components/dash-core-components/src/components/Loading.react.js

```diff
@@ -7,66 +7,130 @@ import CircleSpinner from '../fragments/Loading/spinners/CircleSpinner.jsx';
 import DotSpinner from '../fragments/Loading/spinners/DotSpinner.jsx';
 import {mergeRight} from 'ramda';
 
-function getSpinner(spinnerType) {
-    switch (spinnerType) {
-        case 'graph':
-            return GraphSpinner;
-        case 'cube':
-            return CubeSpinner;
-        case 'circle':
-            return CircleSpinner;
-        case 'dot':
-            return DotSpinner;
-        default:
-            return DefaultSpinner;
-    }
-}
-
-const hiddenContainer = {visibility: 'hidden', position: 'relative'};
-
-const coveringSpinner = {
-    visibility: 'visible',
-    position: 'absolute',
-    top: '0',
-    height: '100%',
-    width: '100%',
-    display: 'flex',
-    justifyContent: 'center',
-    alignItems: 'center',
+const spinnerComponentOptions = {
+    graph: GraphSpinner,
+    cube: CubeSpinner,
+    circle: CircleSpinner,
+    dot: DotSpinner,
 };
 
+const getSpinner = spinnerType =>
+    spinnerComponentOptions[spinnerType] || DefaultSpinner;
+
 /**
  * A Loading component that wraps any other component and displays a spinner until the wrapped component has rendered.
  */
-export default class Loading extends Component {
-    render() {
-        const {
-            loading_state,
-            color,
-            className,
-            style,
-            parent_className,
-            parent_style,
-            fullscreen,
-            debug,
-            type: spinnerType,
-        } = this.props;
-
-        const isLoading = loading_state && loading_state.is_loading;
-        const Spinner = isLoading && getSpinner(spinnerType);
-
-        return (
-            <div
-                className={parent_className}
-                style={
-                    isLoading
-                        ? mergeRight(hiddenContainer, parent_style)
-                        : parent_style
+const Loading = ({
+    children,
+    loading_state,
+    mode,
+    color,
+    className,
+    style,
+    parent_className,
+    parent_style,
+    overlay_style,
+    fullscreen,
+    debug,
+    show_initially,
+    type: spinnerType,
+    delay_hide,
+    delay_show,
+    target_components,
+    custom_spinner,
+}) => {
+    const coveringSpinner = {
+        visibility: 'visible',
+        position: 'absolute',
+        top: '0',
+        height: '100%',
+        width: '100%',
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+    };
+    const hiddenContainer = mergeRight(
+        {visibility: 'hidden', position: 'relative'},
+        overlay_style
+    );
```

Comment:
I thought about that, but I expect most people would tweak the opacity and background color  depending on the theme (light/dark).   Or do know some option that would work in most cases?

Reply:
Could probably pick a neutral grey? I've found `#7F8487` is a good middle ground grey.

Reply:
I can try that - but still want to give the ability for people to use any CSS in `overlay_style`.  Are you suggesting another prop, perhaps  that sets `{"visibility":"visible", "opacity": .5, "backgroundColor": "#7F8487"}`?
Then what if people set both props?

If there are a couple examples in the docs, it might work OK to just leave it as-is?

Reply:
Yeah, maybe let's leave as-is for now with good examples in the docs, and see if there is community feedback from there?